platform-demo-module/src/test/java/com/softicar/platform/demo/module/invoice/stale/StaleDemoInvoiceRefreshTest.java

```diff
@@ -0,0 +1,140 @@
+package com.softicar.platform.demo.module.invoice.stale;
+
+import com.softicar.platform.common.date.Day;
+import com.softicar.platform.common.date.DayTime;
+import com.softicar.platform.core.module.transaction.AGTransaction;
+import com.softicar.platform.core.module.user.CurrentUser;
+import com.softicar.platform.demo.module.AbstractDemoModuleTest;
+import com.softicar.platform.demo.module.invoice.AGDemoInvoice;
+import com.softicar.platform.dom.elements.testing.node.tester.DomNodeTester;
+import com.softicar.platform.emf.EmfMarker;
+import com.softicar.platform.emf.management.EmfManagementDivBuilder;
+import com.softicar.platform.emf.test.tester.EmfFormPopupTester;
+import org.junit.Test;
+
+public class StaleDemoInvoiceRefreshTest extends AbstractDemoModuleTest {
```

Comment:
this is the core of this PR, it tests the behavior in case of concurrent modifications